extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -93,17 +98,39 @@ export abstract class AzureAuth implements vscode.Disposable {
 		let loginComplete: Deferred<void, Error>;
 		try {
 			Logger.verbose('Starting login');
-			const result = await this.login(this.commonTenant, this.metadata.settings.microsoftResource);
-			loginComplete = result.authComplete;
-			if (!result?.response) {
-				Logger.error('Authentication failed');
-				return {
-					canceled: false
+			if (this.authLibrary === 'ADAL') {
+				const result = await this.login(this.commonTenant, this.metadata.settings.microsoftResource);
+				loginComplete = result.authComplete;
+				if (!result?.response) {
+					Logger.error('Authentication failed');
+					return {
+						canceled: false
+					};
+				}
+				const account = await this.hydrateAccount(result.response.accessToken, result.response.tokenClaims);
+				loginComplete?.resolve();
+				return account;
+			} else {
+				const result = await this.loginMsal(this.commonTenant, this.metadata.settings.microsoftResource);
+				loginComplete = result.authComplete;
+				if (!result?.response) {
+					Logger.error('Authentication failed');
+					return {
+						canceled: false
+					};
+				}
+				const token: Token = {
+					token: result.response.accessToken,
+					key: result.response.account.homeAccountId,
+					tokenType: result.response.tokenType
 				};
+				// build token claims object
```

Comment:
How is this "building" anything? Looks like you're just taking it directly from the response

Reply:
this is just what was there before, I just moved it into an else statement.  I can get rid of this comment though if you don't want it

Reply:
I'll remove it